stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/WebSocketClientHandshakeListener.java

```diff
@@ -84,17 +108,84 @@ public void onError(Throwable throwable, HttpCarbonResponse response) {
         }
         ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_HTTP_PKG_ID,
                                                                            WebSocketConstants.WEBSOCKET_CONNECTOR);
-        WebSocketConnectionInfo connectionInfo = getWebSocketOpenConnectionInfo(null, webSocketConnector);
-        countDownLatch.countDown();
-        WebSocketResourceDispatcher.dispatchOnError(connectionInfo, throwable);
+        WebSocketConnectionInfo connectionInfo = getWebSocketOpenConnectionInfo(null,
+                webSocketConnector, webSocketClient, wsService);
+        if (countDownLatch != null) {
+            countDownLatch.countDown();
+        }
+        if (WebSocketUtil.hasRetryConfig(webSocketClient) && throwable instanceof IOException &&
+                WebSocketUtil.reconnect(connectionInfo)) {
+            return;
+        }
+        dispatchOnError(connectionInfo, throwable);
     }
 
-    private WebSocketConnectionInfo getWebSocketOpenConnectionInfo(WebSocketConnection webSocketConnection,
-                                                                   ObjectValue webSocketConnector) {
+    private static WebSocketConnectionInfo getWebSocketOpenConnectionInfo(WebSocketConnection webSocketConnection,
+                                                                         ObjectValue webSocketConnector,
+                                                                         ObjectValue webSocketClient,
+                                                                         WebSocketService wsService) {
         WebSocketConnectionInfo connectionInfo = new WebSocketConnectionInfo(
                 wsService, webSocketConnection, webSocketClient);
         webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
         webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
         return connectionInfo;
     }
+
+    private static void setWebSocketClient(ObjectValue webSocketClient, HttpCarbonResponse carbonResponse,
+                                           WebSocketConnection webSocketConnection, RetryContext retryConfig) {
+        //Using only one service endpoint in the client as there can be only one connection.
+        webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(carbonResponse));
+        if (retryConfig != null && retryConfig.isFirstConnectionMadeSuccessfully()) {
+            webSocketClient.set(WebSocketConstants.LISTENER_ID_FIELD, webSocketConnection.getChannelId());
+        } else {
+            WebSocketUtil.populateWebSocketEndpoint(webSocketConnection, webSocketClient);
+        }
+    }
+
+    /**
+     * Call the readNextFrame().
+     *
+     * @param readyOnConnect ready on connect
```

Comment:
```suggestion
     * @param readyOnConnect ready on connect
```
Need more explanation.